src/libsystemd/sd-bus/bus-objects.c

```diff
@@ -1610,6 +1611,102 @@ static int vtable_member_compare_func(const struct vtable_member *x, const struc
 
+typedef enum {
+        NAMES_FIRSTPART         = 1 << 0, /* first part of argument name list (input names). It is reset by names_are_valid() */
+        NAMES_PRESENT           = 1 << 1, /* at least one argument name is present, so the names will checked.
+                                             This flag is set and used internally by names_are_valid(), but needs to be stored across calls for 2-parts list  */
+        NAMES_SINGLEPART        = 1 << 2, /* argument name list consisting of a single part */
+} names_flags;
+
+static bool names_are_valid(const char *signature, const char **names, names_flags *flags) {
+        int r;
+
+        if ((*flags & NAMES_FIRSTPART || *flags & NAMES_SINGLEPART) && **names != '\0')
+                *flags |= NAMES_PRESENT;
+
+        for (;*flags & NAMES_PRESENT;) {
+                size_t l;
+
+                if (!*signature)
+                        break;
+
+                r = signature_element_length(signature, &l);
+                if (r < 0)
+                        return false;
+
+                if (**names != '\0') {
+                        if (!member_name_is_valid(*names))
+                                return false;
+                        *names += strlen(*names) + 1;
+                } else if (*flags & NAMES_PRESENT)
+                        return false;
+
+                signature += l;
+        }
+        /* let's check if there are more argument names specified than the signature allows */
+        if (*flags & NAMES_PRESENT && **names != '\0' && !(*flags & NAMES_FIRSTPART))
+                return false;
+        *flags &= ~NAMES_FIRSTPART;
+        return true;
+}
+
+/* the current version of this struct is defined in sd-bus-vtable.h, but we need to list here the historical versions
+   to make sure the calling code is compatible with one of these */
+struct sd_bus_vtable_v241 {
+        uint8_t type:8;
+        uint64_t flags:56;
+        union {
+                struct {
+                        size_t element_size;
+                } start;
+                struct {
+                        const char *member;
+                        const char *signature;
+                        const char *result;
+                        sd_bus_message_handler_t handler;
+                        size_t offset;
+                } method;
+                struct {
+                        const char *member;
+                        const char *signature;
+                } signal;
+                struct {
+                        const char *member;
+                        const char *signature;
+                        sd_bus_property_get_t get;
+                        sd_bus_property_set_t set;
+                        size_t offset;
+                } property;
+        } x;
+};
+/* Structure size up to v241 */
+#define VTABLE_ELEMENT_SIZE_241 sizeof(struct sd_bus_vtable_v241)
```

Comment:
referencing systemd versions in comments is fine, but let's avoid this if we don't have to in the main codebase, since downstream distros tend to backport stuff. maybe name this VTABLE_ELEMTN_SIZE_WITHOUT_NAMES or so, i.e. emphasize the change itself rather than the version nr attached to the change

Reply:
and then the next modification will be ..._WITHOUT_SOMETHING_ELSE ?
Ok for me, even if I preferred numbers at least internally. Could be ..._V1, ..._V2, ..._Vn, referring to the table version only.

Reply:
it might make sense to invert the feature thing, i.e. VTABLE_ELEMENT_SIZE_ORIGINAL and then VTABLE_ELEMENT_SIZE_WITH_FOO, VTABLE_ELEMENT_SIZE_WITH_BAR and so on. i.e. be positive rather than negative.

Reply:
being positive is a good thing, and here is a lot of thinking for maybe a single modification.
but... in the unlikely case that in the future several versions are added, then it is difficult to know where we stand, and the applicability of all these labels (numbers included) must remain restricted to this file.